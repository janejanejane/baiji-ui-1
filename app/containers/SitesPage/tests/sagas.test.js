/**
 * Test  sagas
 */

/* eslint-disable redux-saga/yield-effects */
// import { take, call, put, select } from 'redux-saga/effects';
import { put } from 'redux-saga/effects';

import {
  loadSitesSaga,
} from '../sagas';

import {
  loadSitesSuccess,
  loadSitesFailed,
} from '../actions';

describe('loadSites Saga', () => {
  let loadSitesGenerator;

  beforeEach(() => {
    loadSitesGenerator = loadSitesSaga();
    // TODO: Add test
    const testCall = loadSitesGenerator.next().value;
    expect(testCall).toMatchSnapshot();
  });

  describe('OnSuccess', () => {
    it('should dispatch success action on success', () => {
      const fixture = [{
        id: 1,
      }];

      const callDescriptor = loadSitesGenerator.next(fixture).value;
      expect(callDescriptor).toEqual(put(loadSitesSuccess(fixture)));
    });
  });

  describe('OnFailed', () => {
    it('should dispatch failed on failure', () => {
      const fixture = {
        message: 'Error message',
      };

      const callDescriptor = loadSitesGenerator.throw(fixture).value;
      expect(callDescriptor).toEqual(put(loadSitesFailed(fixture.message)));
    });
  });
});
