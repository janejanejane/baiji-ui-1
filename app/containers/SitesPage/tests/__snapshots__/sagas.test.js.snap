// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadSites Saga OnFailed should dispatch failed on failure 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/sites",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`loadSites Saga OnSuccess should dispatch success action on success 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/sites",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
